geomstats/visualization/symmetric_positive_definite.py

```diff
@@ -0,0 +1,598 @@
+"""Functions for SPD Visualization."""
+
+import colorsys
+import math
+
+import matplotlib.pyplot as plt
+import numpy as np
+from geomstats.geometry.spd_matrices import SPDAffineMetric, SPDMatrices
+from geomstats.visualization import Ellipses
+from matplotlib.tri import Triangulation
+from mpl_toolkits.mplot3d.art3d import Poly3DCollection
+
+
+class SymPosDefVisualization:
+    """Class for the visualization of the manifold for SPD matrices.
+
+    This class provides all the essential methods for the
+    visualization of the manifold of the Symmetric Positive
+    Definite matrices.
+
+    Parameters
+    ----------
+    max_z: int
+    The scaling factor of the manifold
+
+    Attributes
+    ----------
+    currZ: int
+    The scaling factor of the manifold
+
+    spdPointViz:
+    Class used to plot points on the manifold SPD(2).
+    Elements S of the manifold of 2D Symmetric Positive
+    Definite matrices can be conveniently represented by ellipses.
+
+    spdManifold:
+    Class for the manifold of symmetric positive definite (SPD) matrices.
+    Takes as input n (int) Integer representing the shape of the matrices: nxn
+    By default n is set to 2 to in order for the visualization to be feasible.
+
+    metric:
+    Compute the affine-invariant exponential map.Compute the Riemannian
+    exponential at point base_point of tangent vector tangent_vec wrt
+    the metric defined in inner_product.
+    This gives a symmetric positive definite matrix.
+
+    References
+    ----------
+    [1] Miolane, Nina, et al. "Geomstats: a Python package for
+    Riemannian geometry in machine learning."
+    Journal of Machine Learning Research 21.223 (2020): 1-9.
+    """
+
+    def __init__(self, max_z=1):
+        """Initialize class parameters of the manifold."""
+        self.max_z = float(max_z)
+        self.currZ = self.max_z
+        self.ax = None
+        self.spdPointViz = Ellipses()
+        self.spdManifold = SPDMatrices(2)
+        self.metric = SPDAffineMetric(2)
+
+    def cuboid_data(self, o, size=(1, 1, 1)):
+        """Generate cuboid data for plotting."""
```

Comment:
missing docstring detailing what are the parameters (se examples of docstring in geomstats).
in particular, what is o?
rename `o` to another, more explicit and readable,, var name